File: frontend/src/dtos/TransacaoDto.ts
export interface TransacaoDto {
    id: number;
    descricao: string;
    valor: number;
    tipo: 'Despesa' | 'Receita';
    pessoaId: number;
    pessoaNome: string;
    categoriaId: number;
    categoriaDescricao: string;
}

export interface TransacaoCriarDto {
    descricao: string;
    valor: number;
    tipo: 1 | 2; // 1 = Despesa, 2 = Receita
    pessoaId: number;
    pessoaNome: string; // enviado vazio, backend preenche
    categoriaId: number;
    categoriaDescricao: string; // enviado vazio, backend preenche
}

File: ControleGastos/ControleGastos/Application/DTOs/TransacaoDto.cs
using ControleGastos.Domain.Enums;

namespace ControleGastos.Application.DTOs;

public class TransacaoDto
{
    public int Id { get; set; }
    public string Descricao { get; set; } = string.Empty;
    public decimal Valor { get; set; }

    public TipoTransacaoEnum Tipo { get; set; }

    public int PessoaId { get; set; }
    public string PessoaNome { get; set; } = string.Empty;

    public int CategoriaId { get; set; }
    public string CategoriaDescricao { get; set; } = string.Empty;
}